src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -683,8 +677,9 @@ public SocketError AcceptAsync(byte[] socketAddress, ref int socketAddressLen, o
 
                 if (operation.TryComplete(this))
                 {
-                    operation.QueueCompletionCallback();
-                    break;
+                    socketAddressLen = operation.SocketAddressLen;
+                    acceptedFd = operation.AcceptedFileDescriptor;
+                    return operation.ErrorCode;
```

Comment:
It's set by the execution of the operation, so it could be Success or any error except IOPending.